script.js

```diff
@@ -1,4 +1,207 @@
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+// Input array of cards and output total score of card
+var countScore = function (cardArray) {
+  var counter = 0;
+  var cardScore = 0;
+  while (counter < cardArray.length) {
+    if (cardArray[counter].rank >= 10) {
+      cardArray[counter].rank = 10;
+    }
+    cardScore = cardScore + cardArray[counter].rank;
+    counter += 1;
+  }
+  return cardScore;
+};
+
+// 6. The user's cards are analysed for winning or losing conditions. Take in arrays of user and computer card. Tally score and output win/lose message.
+var compareWinner = function (userArray, compArray) {
+  var userCardScore = countScore(userArray);
+  var compCardScore = countScore(compArray);
+
+  //Output win/lose message based on scoring matrix
+  var outputMessage = `Player scored ${userCardScore} and Computer scored ${compCardScore}<br><br>`;
+  if (userCardScore >= 22 && compCardScore <= 21) {
+    outputMessage = outputMessage + "Player Busto! You lose!";
+  } else if (compCardScore >= 22 && userCardScore <= 21) {
+    outputMessage = outputMessage + "Computer Busto! You win!";
+  } else if (userCardScore >= 22 && compCardScore >= 22) {
+    outputMessage = outputMessage + "Player and computer Busto! It's a draw!";
+  } else if (userCardScore > compCardScore) {
+    outputMessage = outputMessage + "You scored closer to 21! You win!";
+  } else if (compCardScore > userCardScore) {
+    outputMessage = outputMessage + "Computer scored closer to 21! You lose!";
+  } else if (userCardScore == compCardScore) {
+    outputMessage = outputMessage + "It's a draw!";
+  }
+
+  return outputMessage;
+};
+var convertCardArrayToList = function (cardArray) {
+  var counter = 0;
+  var outputMessage = "";
+  while (counter < cardArray.length) {
+    outputMessage =
+      outputMessage +
+      `<br>${cardArray[counter].name} of ${cardArray[counter].suit}`;
+    counter += 1;
+  }
+  return outputMessage;
+};
+
+//Define global variables
+var gameMode = "game start";
+var shuffledDeck = [];
+var userCards = [];
+var computerCards = [];
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = "Game time! Click submit to deal!";
+  // 2. User clicks Submit to deal 2 cards to each player.
+  if (gameMode == "game start") {
+    // 1. Deck is shuffled.
+    shuffledDeck = shuffleCards(makeDeck());
+    console.log(shuffledDeck);
+    userCards = [];
+    computerCards = [];
+    userCards.push(shuffledDeck.pop());
+    userCards.push(shuffledDeck.pop());
+    computerCards.push(shuffledDeck.pop());
+    computerCards.push(shuffledDeck.pop());
+    console.log(userCards);
+    console.log(computerCards);
+    gameMode = "check and show cards";
+    return myOutputValue;
+  }
+
+  if (gameMode == "check and show cards") {
+    // 3. The cards are analysed for game winning conditions, e.g. Blackjack.
+
+    if (
```

Comment:
you should return at the end of the if / else block instead of after every statement, unless you're using the if/ return format